setup/steps/li_confirm_set_up_auto_explain.go

```diff
@@ -0,0 +1,76 @@
+package steps
+
+import (
+	"errors"
+	"strings"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/guregu/null"
+	"github.com/pganalyze/collector/setup/state"
+	s "github.com/pganalyze/collector/setup/state"
+	"github.com/pganalyze/collector/setup/util"
+)
+
+var ConfirmSetUpAutoExplain = &s.Step{
+	// N.B.: this step, asking the user whether to set up automated explain, is *not* an AutomatedExplainStep
+	// itself, but it is a state.LogInsightsStep because it depends on log insights
+	Kind:        state.LogInsightsStep,
+	Description: "Confirm whether to set up the optional Automated EXPLAIN feature",
+	Check: func(state *s.SetupState) (bool, error) {
+		// skip the question if we've already answered one way or the other
+		if state.Inputs.ConfirmSetUpAutomatedExplain.Valid {
+			return true, nil
+		}
+
+		// also skip the question if it's already set up
+		if state.CurrentSection.HasKey("enable_log_explain") {
```

Comment:
I think there is a bug here, because we return early if the key exists, making the lines that follow not work if the key doesn't exist:

```
* Confirm whether to set up the optional Automated EXPLAIN feature
  ✗ step check failed: error when getting key of section "server1": key "enable_log_explain" not exists
```

I assume the intent was something like this?

```golang
                if state.CurrentSection.HasKey("enable_log_explain") {
                        isLogExplainKey, err := state.CurrentSection.GetKey("enable_log_explain")
                        if err != nil {
                                return false, err
                        }
                        isLogExplain, err := isLogExplainKey.Bool()
                        if err != nil {
                                return false, err
                        }
                        if isLogExplain {
                                return true, nil
                        }
                }

                // assume auto_explain if we got this far
```